### PR TITLE
Added GTDB genome object copying and Tree viz

### PR DESCRIPTION
**Pull request by dcchivian**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -33,6 +33,14 @@ WORKDIR /kb
 RUN git clone https://github.com/marbl/Krona.git && \
     cd Krona/KronaTools/ && \
     ./install.pl
+
+# Install ETE3
+RUN apt-get update && \
+    apt-get -y install xvfb
+RUN pip install --upgrade pip
+# Note: You must use PyQt5==5.11.3 on debian
+RUN pip install ete3==3.1.3 PyQt5==5.11.3 numpy==1.23.1
```
Comment (Tianhao-Gu) [inline Dockerfile:42]: apt-get update, --upgrade pip, and pip install numpy==1.23.1 has already been run.

File: lib/kb_gtdbtk/core/genome_obj_update.py [hunk 2]
```diff
@@ -81,6 +129,621 @@ def check_obj_type_assembly(
...
+    found_output = False
+    
+    time_spent = 0
+    while time_spent <= patience:
+        found_output = True
+        for this_file in target_files:
+            if not os.path.exists(this_file) or not os.path.getsize(this_file) > 0:
+                found_output = False
+                print ("waiting for {} seconds...".format(sleep_interval))
+                time.sleep(sleep_interval)
```
Comment (Tianhao-Gu) [inline lib/kb_gtdbtk/core/genome_obj_update.py:156]: time is not imported.

File: lib/kb_gtdbtk/core/genome_obj_update.py [hunk 3]
```diff
@@ -81,6 +129,621 @@ def check_obj_type_assembly(
...
+    return file_links
+
+
+# get_genome_id_to_upa_map()
+#
+def get_genome_id_to_upa_map (genome_upas_map_file):
+    genome_id_to_upa_map = dict()
+    this_file = genome_upas_map_file
+    if this_file.lower().endswith('.gz'):
+        f = gzip.open(this_file, 'rt')
```
Comment (Tianhao-Gu) [inline lib/kb_gtdbtk/core/genome_obj_update.py:572]: neither does gzip

**Review by Tianhao-Gu — CHANGES_REQUESTED** [review]
I scanned through it for syntax errors only. It becomes really time consuming when there are more than 500 lines of changes.

File: Dockerfile (hunk 1 above)
Comment (dcchivian) [reply Dockerfile]: fixed

File: lib/kb_gtdbtk/core/genome_obj_update.py (hunk 2 above)
Comment (dcchivian) [reply lib/kb_gtdbtk/core/genome_obj_update.py:158]: fixed

File: lib/kb_gtdbtk/core/genome_obj_update.py (hunk 3 above)
Comment (dcchivian) [reply lib/kb_gtdbtk/core/genome_obj_update.py:574]: fixed

**Review by Tianhao-Gu — APPROVED** [review]
👍

**Pull request merged**